uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/leave-request.model.js

```diff
@@ -13,9 +15,35 @@ define([
     'Model',
     'LeaveRequestAPI',
     'LeaveRequestInstance',
-    function ($log, Model, leaveRequestAPI, instance) {
+    'SicknessRequestInstance',
+    'TOILRequestInstance',
+    function ($log, Model, leaveRequestAPI, LeaveRequestInstance,
+      SicknessRequestInstance, TOILRequestInstance) {
       $log.debug('LeaveRequest');
 
+      /**
+       * Gets the instance for the given request fetched via API,
+       * for example for "toil" request that would be `TOILRequestInstance`.
+       * Defaults to the `LeaveRequestInstance`.
+       *
+       * @param  {Object} requestObject a plain LeaveRequest object fetched via API
+       * @return {Instance} instance of the request depending on its type
+       */
+      function initRequestInstance (requestObject) {
+        var requestInstance;
+        var requestType = requestObject.request_type;
+
+        if (requestType === 'sickness') {
```

Comment:
why not use a map instead?

```js
var requestTypeInstance = {
  sickness: SicknessRequestInstance,
  toil: TOILRequestInstance,
  leave: LeaveRequestInstance
};

return requestTypeInstance[requestType].init(requestObject, true);
```

Reply:
if you have this mapping you don't even need the `initRequestInstance` function at all

Reply:
actually forget the mappin, just inject the desired instance type on the fly by using `$injector`
```js
var myService = $injector.get('MyServiceName');
```

Reply:
@reneolivo mapping would not allow default value.
@AkA84 this func is used in 2 places in `all()` and `find()`.

Reply:
@igorpavlov is there a case where the request type is not defined? I don't think so, and it should never be the case.